lib/bindings/python/rust/context.rs

```diff
@@ -15,11 +16,29 @@ use std::sync::Arc;
 #[pyclass]
 pub struct Context {
     inner: Arc<dyn AsyncEngineContext>,
+    trace_context: Option<DistributedTraceContext>,
 }
 
 impl Context {
     pub fn new(inner: Arc<dyn AsyncEngineContext>) -> Self {
-        Self { inner }
+        Self {
+            inner,
+            trace_context: None,
+        }
+    }
+
+    // Create context with trace information
+    #[must_use]
+    pub fn with_trace_context(
+        inner: Arc<dyn AsyncEngineContext>,
+        trace_context: Option<DistributedTraceContext>
```

Comment:
maybe we can remove the `Option<...>` wrapper, because if the user is not providing a `trace_context`, the `new` method should be used instead?

Reply:
maybe we can combine this constructor with the existing `new` constructor, and update calls into `new` to pass in a `None` as the second argument?

Reply:
combined